lib/src/repository/traits/thermostat_repository.dart

```diff
@@ -25,6 +25,14 @@ class ThermostatRepository {
           new Set<AvailableThermostatMode>.from(
               trait.properties.availableThermostatModes);
 
+      final TemperatureRange coolTempRange = new TemperatureRange(
+          min: trait.properties.coolSetPointRange?.min ?? -double.maxFinite,
```

Comment:
This is an alternative to what I had changed before where I was using a null TemperatureRange object that would be functionally equivalent I think. We would have a temperature range available but in validation it would be the entire range of the double type so it would effectively be no validation when platform returns a null range.